# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/PuppetWidget.cpp
# Commit: 0df2f2832755
# Full Hash: 0df2f283275530fbfe5cf9c95986b073c0d6a31e
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-23 02:46:57
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 1: Use BigBuffer for IPCDataTransfer, r=edgar
#   
#   The IPCDataTransfer type is used to transfer Clipboard/Drag & Drop payloads
#   over IPC to allow them to be written to or read from the relevant system
#   interfaces. Previously, the system which was used was somewhat complex, and
# ==============================================================================

diff -r 1d21d911b3e7 -r 0df2f2832755 widget/PuppetWidget.cpp
--- a/widget/PuppetWidget.cpp	Mon Aug 22 15:35:28 2022 +0000
+++ b/widget/PuppetWidget.cpp	Mon Aug 22 15:38:14 2022 +0000
@@ -897,7 +897,7 @@
   }
 
   bool hasCustomCursor = false;
-  UniquePtr<char[]> customCursorData;
+  Maybe<mozilla::ipc::BigBuffer> customCursorData;
   size_t length = 0;
   IntSize customCursorSize;
   int32_t stride = 0;
@@ -935,10 +935,8 @@
     }
   }
 
-  nsDependentCString cursorData(customCursorData ? customCursorData.get() : "",
-                                length);
   if (!mBrowserChild->SendSetCursor(
-          aCursor.mDefaultCursor, hasCustomCursor, cursorData,
+          aCursor.mDefaultCursor, hasCustomCursor, std::move(customCursorData),
           customCursorSize.width, customCursorSize.height, resolution.mX,
           resolution.mY, stride, format, aCursor.mHotspotX, aCursor.mHotspotY,
           force)) {